flink-metrics/flink-metrics-statsd/src/test/java/org/apache/flink/metrics/statsd/StatsDReporterTest.java

```diff
@@ -262,6 +262,69 @@ public void testStatsDMetersReporting() throws Exception {
 		}
 	}
 
+	/**
+	 * Tests that meters are properly reported via the StatsD reporter.
+	 */
+	@Test
+	public void testStatsDMetersReportingOfNegativeValues() throws Exception {
+		MetricRegistryImpl registry = null;
+		DatagramSocketReceiver receiver = null;
+		Thread receiverThread = null;
+		long timeout = 5000;
+		long joinTimeout = 30000;
+
+		String meterName = "meter";
+
+		try {
+			receiver = new DatagramSocketReceiver();
+
+			receiverThread = new Thread(receiver);
+
+			receiverThread.start();
+
+			int port = receiver.getPort();
+
+			MetricConfig config = new MetricConfig();
+			config.setProperty(ConfigConstants.METRICS_REPORTER_INTERVAL_SUFFIX, "1 SECONDS");
+			config.setProperty("host", "localhost");
+			config.setProperty("port", String.valueOf(port));
+
+			registry = new MetricRegistryImpl(
+				MetricRegistryConfiguration.defaultMetricRegistryConfiguration(),
+				Collections.singletonList(ReporterSetup.forReporter("test", config, new StatsDReporter())));
```

Comment:
You can skip the creation of the registry. You can extract the reporter from the ReporterSetup and call report explicitly.

Reply:
Done, it's a lot cleaner now. I've adjusted the other unit tests in the class like this as well